### PR TITLE
DSD-1839: Prevent link with icon line break

### PR DESCRIPTION
**Pull request by 7emansell**
Fixes JIRA ticket [DSD-1839](https://newyorkpubliclibrary.atlassian.net/browse/DSD-1839)

## This PR does the following:

- Adds `whiteSpace: nowrap` to the `moreLink` variant 
   - Unrelated, but explicitly types the `icon` theme file since it was "exceeding compiler size"

## How has this been tested?

Storybook

### Accessibility Checklist

- [x] Checked Storybook's "Accessibility" tab for color contrast and other issues.
- [x] The feature works with keyboard inputs including tabbing back and forward and pressing space, enter, arrow, and esc keys.
- [x] For hidden text or when `aria-live` is used, a screenreader was used to verify the text is read.
- [x] For features that involve UI updates and focusing on DOM `ref`s, focus management was reviewed.
- [x] The feature works when the page is zoomed in to 200% and 400%.

### Checklist:

- [x] I have updated the Storybook documentation and changelog accordingly.
- [x] I have added relevant accessibility documentation for this pull request.
- [x] All new and existing tests passed.

### Front End Review:

- [x] Review the Vercel preview deployment once it is ready.

[DSD-1839]: https://newyorkpubliclibrary.atlassian.net/browse/DSD-1839?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by oliviawongnyc — APPROVED** [review]

File: src/theme/components/link.ts [hunk 1]
```diff
@@ -52,6 +52,7 @@ const moreLink = definePartsStyle(
     base: {
       alignItems: "center",
       display: "inline-flex",
+      whiteSpace: "nowrap",
```
Comment (bigfishdesign13) [inline src/theme/components/link.ts:55]: Why did you add this to just the `moreLink` variant and not to all variants? Icons can be passed into any variant.

File: src/theme/components/icon.ts [hunk 2]
```diff
@@ -1,18 +1,25 @@
...
+  height: string;
+} = {
   display: "inline-block",
   fill: "currentColor",
   width: "auto",
   height: "100%",
 };
-const align = {
+
+const align: Record<string, { marginEnd?: string; marginStart?: string }> = {
```
Comment (bigfishdesign13) [inline src/theme/components/icon.ts:16]: Why was `Record` added here and elsewhere?

**Comment by EdwinGuzman** [comment]
> Unrelated, but explicitly types the icon theme file since it was "exceeding compiler size"

Where did you see this?

File: src/theme/components/link.ts (hunk 1 above)
Comment (7emansell) [reply src/theme/components/link.ts:55]: I wanted to fix the styling on the ones that the DS controls– it felt like if you're adding your own icon, it's your responsibility to style it as necessary?

**Comment by 7emansell** [comment]
> > Unrelated, but explicitly types the icon theme file since it was "exceeding compiler size"
> 
> Where did you see this?

Build error

File: src/theme/components/icon.ts (hunk 2 above)
Comment (7emansell) [reply src/theme/components/icon.ts:16]: see [here](https://github.com/NYPL/nypl-design-system/pull/1694#issuecomment-2457362609)

Comment (EdwinGuzman) [reply src/theme/components/icon.ts:16]: I'm surprised we don't see more of these build errors since this pattern is used elsewhere.

**Review by EdwinGuzman — APPROVED** [review]

File: src/theme/components/link.ts (hunk 1 above)
Comment (bigfishdesign13) [reply src/theme/components/link.ts:55]: Ok, makes sense.

**Review by bigfishdesign13 — APPROVED** [review]
Looks good!
